codeql.groovy

```diff
@@ -0,0 +1,50 @@
+#!./lib/runner.groovy
+// Generates server-side metadata for Gradle auto-installation
+import com.gargoylesoftware.htmlunit.Page
+import com.gargoylesoftware.htmlunit.WebClient
+import com.gargoylesoftware.htmlunit.WebResponse
+
+import net.sf.json.*
+
+def wc = new WebClient();
+wc.addRequestHeader("accept", "application/json")
+wc.getOptions().setThrowExceptionOnFailingStatusCode(false);
+wc.getOptions().setPrintContentOnFailingStatusCode(false);
+
+def API_URL = "https://api.github.com";
+Page p = wc.getPage(API_URL + "/repos/github/codeql-action/releases")
```

Comment:
This looks brittle without API token being used

Reply:
@daniel-beck a couple of [other crawlers](https://github.com/jenkins-infra/crawler/search?q=api) seem to do the same thing but happy to change it if it makes sense! Are there any GitHub API keys that get injected to the build by chance?

Reply:
Oh wow, I didn't know that we're already doing that.

Which is kind of even worse with the (very low IIRC) anonymous API limits of GH, the more crawlers use them, the more likely it is we hit the limit.

Not sure about a credential. Maybe something to do later if it doesn't work anymore after all.

Reply:
it may not even be rate limited. releases in general is CDN backed

Reply:
if thats the case, can I get a 👍 😉